packages/gator-permissions-snap/src/clients/accountApiClient.ts

```diff
@@ -70,6 +117,52 @@ export class AccountApiClient {
     return chainId === 1;
   }
 
+  /**
+   * Makes a secure HTTP request with timeout and response size limits.
+   * @param url - The URL to fetch.
+   * @returns A promise that resolves to the response.
+   * @throws {ResourceUnavailableError} If the request times out or exceeds size limits.
+   */
+  async #makeSecureRequest(url: string): Promise<globalThis.Response> {
```

Comment:
nit: the word "secure" is a little misleading imo. I think it's added to indicate that there are mitigations, such as timeout, max response size header, etc. But I think in the context of a request, it implies that it enforces HTTPS.

Reply:
Also, this method is duplicated in priceApiClient.

Could we decompose it out, and maybe accept the zod response definition as a parameter so that it makes the request with all of the mitigations _including_ validating the response against the Zod type?

Could either be a static function, or a base class that includes the timeout, max response size etc configuration.